### PR TITLE
Add mu installation note for OSX/Homebrew users

### PR DESCRIPTION
**Pull request by Karunamon**
I discovered this after a good deal of messing about trying to get mu to work, and figured it may be useful to include in the official docs.

### CODE REVIEW

File: layers/+email/mu4e/README.org [hunk 1]
```diff
@@ -19,6 +19,14 @@
 In order to use this layer you must install mu and mu4e separately. Typically
 mu4e will be bundled with mu (this is the case on many Linux distributions).
 
+If you're on OS X and install mu using Homebrew, you must specify the
+location of your Emacs binary at install time using the EMACS environment
+variable, as well as passing the --with-emacs option:
+
+#+begin_src shell
+EMACS=$(which emacs) brew install homebrew --with-emacs
```
Comment (d12frosted) [inline layers/+email/mu4e/README.org]: Hey, what is the meaning of this?

Comment (d12frosted) [inline layers/+email/mu4e/README.org]: Did you just mean `brew install mu --with-emacs`?

**Comment by d12frosted** [comment]
Hey, thanks for PR. It's a pity to hear that you struggled with this. 

I am not sure that `EMACS` environment variable is required. Though you have to install `mu` using `brew install mu --with-emacs` and then setup `mu4e-installation-path` as it mentioned in readme file. For you it should be something like `"/usr/local/Cellar/mu/0.9.16/share/emacs/site-lisp/"` with respect to latest version.

**Comment by Karunamon** [comment]
Heh, you caught me halfway through writing this up.

The idea here was simplest minimum viable configuration. In the spirit of some of the OSX specifics sprinkled throughout the documentation, and the fact that just installing `mu` without the extra command line options won't give you a working config.

I had to do the `EMACS=` bit or else mu4e would not be correctly linked (I think it was putting it in a path accessible to the system emacs, rather than spacemacs). It also obviates the need to set the `mu4e-installation-path` option.

**Comment by d12frosted** [comment]
Oh, just checked my configs. Totally forgot about another location of `brew` packages that should be used instead of installation path. You might want to use `/usr/local/opt/mu/share/emacs/site-lisp/`.

**Comment by d12frosted** [comment]
> Heh, you caught me halfway through writing this up.

Haha. It happens 😸 💃

**Comment by Karunamon** [comment]
I just ran a test again to confirm this, by backing up my `.emacs.d`, doing a fresh install, and blowing away `mu`.

Steps look like:

1. Install spacemacs
2. Edit `.emacs.d` and add the `mu4e` layer
3. Sync config
4. Attempt to run with: `SPC a m`:  `Cannot open load file: No such file or directory: mu4e`
5. Install `mu` using the `--with-emacs` argument to Homebrew
6. Relaunch spacemacs and attempt running again. **Same error**
7. Uninstall `mu`, reinstall `mu` with `EMACS=$(which emacs) brew install mu --with-emacs`
8. Relaunch spacemacs, attempt running: **It works!**

**Comment by d12frosted** [comment]
I still don't get the point of `EMACS=$(which emacs) brew install mu --with-emacs`. Why is `brew` part in `EMACS` variable? Could you please try setting the value of `mu4e-installation-path` to `/usr/local/opt/mu/share/emacs/site-lisp/`? This is what actually README file asks users to do after installing `mu`. 

Don't get me wrong, I am all for mentioning that users must install `mu` with `brew` using `brew install mu --with-emacs`. But the part about `EMACS=$(which emacs) brew install mu --with-emacs` seems suspicious to me.

**Comment by d12frosted** [comment]
Also, after you install `mu` using `brew install mu` executing `brew install mu --with-emacs` has no effect as it should warn you that `mu` was already installed. You should use `brew reinstall mu --with-emacs`.

**Comment by TheBB** [comment]
> I still don't get the point of `EMACS=$(which emacs) brew install mu --with-emacs`. Why is `brew` part in `EMACS` variable?

What this does is it assigns the output of `which emacs` to the `EMACS` variable for the duration of the command `brew install mu --with-emacs`. So the `brew` is part of the command that runs, not the value being assigned.

**Comment by d12frosted** [comment]
Thanks for explanation, @TheBB. But there is no separator between `EMACS=$(which emacs)` and `brew install mu --with-emacs`. So `EMACS=$(which emacs) brew install mu --with-emacs` actually only sets var. At least if I understand it properly 😸 💃 

But your comment forced me to investigate `mu` build script and it seems that setting `EMACS` environment variable has the same effect as using `--with-emacs` flag in the first place while installing `mu`. So I am against overcomplicating installation instructions. 

If you ever install `brew` package with wrong options, you have to do the following.

```
$ brew uninstall package
$ brew install package [options]
```

Even `reinstall` reuses previously used options, so when you really want to build with different options the best way is to use `uninstall` and `install` 😸 Hope it helps.

**Comment by TheBB** [comment]
No, I think you're misunderstanding. What separator are you looking for? Try it:

```sh
$ TEST=test env | grep TEST
TEST=test
```

**Comment by d12frosted** [comment]
@TheBB, my bad 😿  I really totally misunderstood that expression. Sorry for overall confusion around it. 

Anyway, the part about `mu` build script still stands. You just have to properly install it in the first place.

**Comment by Karunamon** [comment]
>Could you please try setting the value of mu4e-installation-path to /usr/local/opt/mu/share/emacs/site-lisp/? This is what actually README file asks users to do after installing mu.

Tried, and that works after a restart of emacs (just doing an evaluate of the setq isn't enough).

So, apparently, this whole thing comes down to whether you want to modify the `brew install mu` command (which you have to do anyways with the `--with-emacs` flag) and have it Just Work, or set the path in emacs after the install.

Either will lead to a working installation, but the former way seems easier.

**Comment by TheBB** [comment]
I dunno what you two have arrived at, but I am assuming it should be mu that's installed and not homebrew?

**Comment by d12frosted** [comment]
@TheBB Well, I am all for adding a note that OS X and brew users must install `mu` using `brew install mu --with-emacs`. But that `var` set is not required as it has the same effect as setting using `--with-emacs` option. And more over, `EMACS=$(which emacs) brew install homebrew --with-emacs` doesn't work in `fish`, while `brew install mu --with-emacs` is universal.

**Comment by TheBB** [comment]
Okay, I'll follow @d12frosted's suggestion. Also changed "homebrew" to "mu".

Thanks! Cherry-picked in develop. You can safely delete your branch.

**Comment by d12frosted** [comment]
@TheBB haha, indeed, `mu`, not `homebrew` :D

**Pull request closed without merging**